### PR TITLE
Marvel level 

### PR DESCRIPTION
**Pull request by samyakbvs**
- [x] Read and understood [see CONTRIBUTING.md](https://github.com/fossasia/labyrinth/blob/master/CONTRIBUTING.md)
- [x] Included a preview link and screenshot showing after and before the changes.
- [x] Included a description of change below.
- [x] Tried to [squash the commits][squash].
- [x] Added Screenshot / Path to tile to Indicate Location of Change in Game

### Changes done in this Pull Request

I am adding a Marvel level to this game wherein the player has to collect all six infinity stones and reach the Infinity gauntlet tile to advance to the next level.

- View my work at https://pr-1066-fossasia-labyrinth.surge.sh

[squash]: https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

### CODE REVIEW

**Comment by samyakbvs** [comment]
@YashKumarVerma I will take care of that, but can you please tell me why my level is not appearing in the preview link

**Comment by YashKumarVerma** [comment]
> @YashKumarVerma I will take care of that, but can you please tell me why my level is not appearing in the preview link

it is because the javascript code you have written has errors, therefore the linting check is not passed.

Since the test is not passed (check build logs on Travis) the deploy command won't run and hence it's not showing the changes.

**Comment by samyakbvs** [comment]
Please merge this PR

**Comment by samyakbvs** [comment]
@YashKumarVerma on the first level, after collecting the sword, I went to the console and entered `player.inventory.has("Sword")`, which returned true, but when I am on the marvel level the same code returns false, even if the sword is still present in the inventory. Can you explain why this is happening?

**Comment by samyakbvs** [comment]
Due to this problem, I am not able to advance the user to the next level, when he reaches the infinity tile with all six stones.

File: js/tiles.js [hunk 1]
```diff
@@ -1716,7 +1716,7 @@ const marvel = {
 
       visit: function() {
 
-        if (player.inventory.has('Soul') ) {
+        if (true) {
         player.addReachableLevel(createFirstLevel());
        } else {
            alertInfo("Get all the six infinty stones to advance to the next level...");
```
Comment (YashKumarVerma) [inline js/tiles.js]: why is there an else associated with always true statement?

**Comment by YashKumarVerma** [comment]
> @YashKumarVerma on the first level, after collecting the sword, I went to the console and entered `player.inventory.has("Sword")`, which returned true, but when I am on the marvel level the same code returns false, even if the sword is still present in the inventory. Can you explain why this is happening?

did you try consoling out the following events 
- when user not entered desired tile with item
- when user enters tile with item
- when user leaves the tile with item
- when you manually add item to inventory from console
- then view player object

this might be helpful.

**Comment by samyakbvs** [comment]
Yes I will remove the files when I have made all the commits @YashKumarVerma

File: js/tiles.js (hunk 1 above)
Comment (samyakbvs) [reply js/tiles.js]: This statement won't be true always

**Comment by samyakbvs** [comment]
@YashKumarVerma all the six infinity stones show up in the user's inventory both in the game and on the console still `player.inventory.has('Soul') && player.inventory.has('Mind') && player.inventory.has('Power') && player.inventory.has('Space') && player.inventory.has('Time') && player.inventory.has('Reality')` retruns false

**Comment by YashKumarVerma** [comment]
> @YashKumarVerma all the six infinity stones show up in the user's inventory both in the game and on the console still `player.inventory.has('Soul') && player.inventory.has('Mind') && player.inventory.has('Power') && player.inventory.has('Space') && player.inventory.has('Time') && player.inventory.has('Reality')` returns false

what do the individual commands like  `player.inventory.has('Power')` return for all stones?

**Comment by samyakbvs** [comment]
Screenshot 2019-12-04 at 3 45 40 PM
After consoling out for a while, I realised that the last stone collected will return true for `player.inventory.has(Stone)` however rest of the stones will return false. For example, if I collected the Mind stone and then run `player.inventory.has("Mind")` it would return true however if I collect the Space stone then player.inventory.has("Space") would return true, but `player.inventory.has("Mind")` would return false. 

Do you why this problem is occurring @YashKumarVerma

**Comment by samyakbvs** [comment]
@YashKumarVerma can you pls merge now, I am not able to squash commits as they are not consecutive.

**Comment by YashKumarVerma** [comment]
These are too many commits to be merged directly into master.

**Comment by samyakbvs** [comment]
I will try to merge all the consecutive ones

**Comment by samyakbvs** [comment]
@YashKumarVerma I believe all the requested changes have been made, so pls merge this PR now.

**Comment by samyakbvs** [comment]
Thanks, @YashKumarVerma so are you merging it then?

**Comment by YashKumarVerma** [comment]
A lot of effort was done in creating and designing this level, along with the functionality which is added like collecting all stones to enter is highly appreciated. The level uses the inventory API beautifully to create an interactive level which is fun to play as well.

The work is great, and considering all these factors the ***Level is being added***. The minor bug which you are facing will be discussed in a separate issue (*it'd be great if you author it*).

# :tada: Good Work


*note for GCI mentors: kindly consider the efforts of the participant to make this PR a success, aka: `brownie points`*

**Comment by samyakbvs** [comment]
I fixed the minor bug as well @YashKumarVerma, thank you!!

**Comment by YashKumarVerma** [comment]
I'll merge it as soon as you change the commit name from ***jiji*** to something else. :laughing:

**Comment by samyakbvs** [comment]
Sure 😆

**Comment by samyakbvs** [comment]
Done @YashKumarVerma

**Comment by YashKumarVerma** [comment]
> Done @YashKumarVerma

there are still 2 commits :confused:

**Comment by samyakbvs** [comment]
> > Done @YashKumarVerma
> 
> there are still 2 commits 😕

Now there is one, @YashKumarVerma

**Pull request merged**